src/lib/Microsoft.Health.Extensions.Fhir.R4/Repository/FhirServiceRepository.cs

```diff
@@ -0,0 +1,129 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using IFhirClient = Microsoft.Health.Fhir.Client.IFhirClient;
+
+namespace Microsoft.Health.Extensions.Fhir.Repository
+{
+    public class FhirServiceRepository : IFhirServiceRepository
```

Comment:
>Repository [](http://example.com/codeflow?start=28&length=10)

Not sure repository is the correct term.  Usually a repository is responsible for retrieving and persisting data from another service.  These are really extensions on top of the existing FHIR Client.  Suggestion to just do these as extension methods.

Reply:
With this PR, we would no longer pass around the FhirClient, so there would be no object to have extension methods against. The original idea was to provide a layer of abstraction between our code and how we communicate between the FhirServer, and to remove hard dependencies to the HL7 Fhir Client. Our code would now retrieve and persist data via this class and hide the fact that we use the FhirClient. 

Do you feel that this level of abstraction is not needed? I do see that the new FhirClient makes use of an interface `IFhirClient`. Should we make use of that everywhere and then make this class extension methods?

Reply:
Agreed that repository may not be the best term. I think a central point of interaction with the FhirClient could provide further extensibility and organization though. An example off the top of my head is if we want to refactor the FhirClient again in the future, we'd only have to update the Repository class

Just my perspective. Looking forward to hearing both your thoughts

Reply:
Sounds good. Thanks Rob and Dustin for your feedback on this. I'm resolving this conversation thread but feel free to reopen if there any further questions/comments